tools/devtool

```diff
@@ -1163,28 +1163,47 @@ cmd_build_kernel() {
     create_dir "$kernel_dir_host"
 
     # Extract the kernel version from the config file provided as parameter.
-    KERNEL_VERSION=$(grep -Po "^# Linux\/$cfg_pattern (([0-9]+.)[0-9]+)" "$KERNEL_CFG" | cut -d ' ' -f 3)
-    validate_kernel_version "$KERNEL_VERSION"
+    kernel_version=$(grep -Po "^# Linux\/$cfg_pattern (([0-9]+.)[0-9]+)" "$kernel_config" | cut -d ' ' -f 3)
+    validate_kernel_version "$kernel_version"
+
+    kernel_major_version="${kernel_version%%.*}"
+    kernel_url_base="https://cdn.kernel.org/pub/linux/kernel"
+    kernel_latest_version=$( \
+        curl -s $kernel_url_base/v$kernel_major_version.x/ \
+        | grep -o "linux-$kernel_version\.[0-9]*\.tar.xz" \
+        | sort -rn -t . -k 3 \
+	| head -n 1)
+    kernel_url="$kernel_url_base/v$kernel_major_version.x/$kernel_latest_version"
 
-    recipe_commit="b551cccc405a73a6d316c0c09dfe0b3e7a73ba3f"
-    recipe_url="https://raw.githubusercontent.com/rust-vmm/vmm-reference/$recipe_commit/resources/kernel/make_kernel.sh"
+    echo "Downlaoding kernel from $kernel_url"
     run_devctr \
         --user "$(id -u):$(id -g)" \
         --workdir "$kernel_dir_ctr" \
-        -- /bin/bash -c "curl -LO $recipe_url && source make_kernel.sh && extract_kernel_srcs $KERNEL_VERSION"
-    ok_or_die "Could not extract kernel sources with recipe $recipe_url!"
-
-    cp "$KERNEL_CFG" "$kernel_dir_host/linux-$KERNEL_VERSION/.config"
-
-    KERNEL_BINARY_NAME="vmlinux-$KERNEL_VERSION-$arch.bin"
+        -- /bin/bash -c "curl $kernel_url > $kernel_latest_version"
+    ok_or_die "Could not download kernel: $kernel_url"
+
+    echo "Extarcting kernel sources..."
+    kernel_dest_dir="linux-$kernel_version"
+    kernel_dest_dir_host="$kernel_dir_host/$kernel_dest_dir"
+    kernel_dest_dir_ctr="$kernel_dir_ctr/$kernel_dest_dir"
+    create_dir "$kernel_dest_dir_host"
     run_devctr \
         --user "$(id -u):$(id -g)" \
         --workdir "$kernel_dir_ctr" \
-        -- /bin/bash -c "source make_kernel.sh && make_kernel "$kernel_dir_ctr/linux-"$KERNEL_VERSION"" $format $target $nprocs $KERNEL_BINARY_NAME"
-    ok_or_die "Could not build kernel!"
+        -- /bin/bash -c "tar --skip-old-files --strip-components=1 -xf $kernel_latest_version -C $kernel_dest_dir"
+    ok_or_die "Could not extract kernel sources: $kernel_dir_host/$kernel_latest_version"
```

Comment:
Since we are extracting the kernel sources in the container, won't they be lost after the container exits? I think it would be simpler to extract all this functionality out of `devtool` to a separate script. Out of scope for now though.

Reply:
Since `<firecrakcer_root>/build/rootfs` is mounted in the container, it won't be lost. But I agree that move them to a separate script. As it seems to not be so difficult, I'll try it. Thanks.

Reply:
Done. Please see `resources/tests/build_kernel.sh`.